pytest_splunk_addon/kubernetes_helper.py

```diff
@@ -0,0 +1,528 @@
+        namespace_name = Name of the namespace
+        """
+        try:
+            wait_count = 0
+            SLEEP_TIME = 30
+            pod_created = False
+            while wait_count <= 5:
+                timer = SLEEP_TIME * wait_count
+                initial_timer = 0
+                while initial_timer < timer:
+                    LOGGER.info("wait_count is {0}".format(str(wait_count)))
+                    api_instance = client.CoreV1Api()
+                    api_response = api_instance.read_namespaced_pod_status(
+                        name=pod_name, namespace=namespace_name
+                    )
+                    if api_response.status.phase != "Pending":
+                        try:
+                            api_response_log = api_instance.read_namespaced_pod_log(
+                                name=pod_name, namespace=namespace_name
+                            )
+                            with open(
+                                "{0}.log".format(pod_name), "w"
+                            ) as splunk_standalone:
+                                splunk_standalone.write(api_response_log)
+                        except Exception as e:
+                            LOGGER.error(
+                                "Found exception in writing the logs for pod : {0}".format(
+                                    pod_name
+                                )
+                            )
+                        # readiness probe
+                        if api_response.status.container_statuses[0].ready == True:
+                            LOGGER.info("Pod {0} created....".format(pod_name))
+                            pod_created = True
+                            break
+                        else:
+                            LOGGER.info(
+                                "waiting for pod {0} to get created...".format(pod_name)
+                            )
+                            initial_timer += 1
+                            time.sleep(1)
+                            continue
+                    else:
+                        LOGGER.warning(
+                            "Pod {0} is still in Pending state.".format(pod_name)
+                        )
+                        initial_timer += 1
+                        time.sleep(1)
+                        continue
+                if pod_created == True:
+                    break
+                wait_count += 1
+            if (wait_count > 5) and (pod_created == False):
+                LOGGER.error(
+                    "Waiting for pod {0} creation took more than expected, please check for logs".format(
+                        pod_name
+                    )
+                )
+        except Exception as e:
+            LOGGER.error(
+                "Found exception while waiting for pod {0} : {1}".format(pod_name, e)
+            )
+
+    def wait_for_deployment_to_get_available(self, deployment_name, namespace_name):
```

Comment:
This function looks super similar to `wait_for_statefulset_to_get_available` and some other functions under this class.